oio/xcute/common/worker.py

```diff
@@ -80,16 +80,13 @@ def _reply(self, reply_addr, reply_tube,
             'task_errors': task_errors
         })
 
-        sender_key = (reply_addr, reply_tube)
-        if sender_key not in self.beanstalkd_senders:
-            self.beanstalkd_senders[sender_key] = BeanstalkdSender(
-                addr=reply_addr,
-                tube=reply_tube,
-                logger=self.logger)
+        beanstalkd_reply_info = (reply_addr, reply_tube)
+        beanstalkd_reply = self.beanstalkd_replies.get(beanstalkd_reply_info)
+        if not beanstalkd_reply:
+            beanstalkd_reply = Beanstalk.from_url(beanstalkd_addr)
+            beanstalkd_reply.use(self.beanstalkd_workers_tube)
+            beanstalkd_reply.watch(self.beanstalkd_workers_tube)
 
-        beanstalkd_sender = self.beanstalkd_senders[(reply_addr, reply_tube)]
+            self.beanstalkd_replies[beanstalkd_reply_info] = beanstalkd_reply
 
-        while not beanstalkd_sender.send_job(reply_payload):
-            sleep(1)
-
-        beanstalkd_sender.job_done()
+        beanstalkd_reply.put(reply_payload)
```

Comment:
What happens when the `put` fails? Does it raise an exception? If so, what does the event-agent do?

Reply:
If this `put` fails, we catch exception here: https://github.com/open-io/oio-sds/pull/1953/files#diff-8cd3ec0f8415e064fe679869ed2e03a8R52

Reply:
Otherwise, if `XcuteFilter.process` raises an exception, the event will be buried (this is the behavior of the event-agent).

Reply:
:+1: